aea/configurations/base.py

```diff
@@ -396,13 +439,137 @@ def __lt__(self, other):
+    def package_prefix(self) -> Tuple[ConfigurationType, str, str]:
+        """Get the package identifier without the version."""
+        return self.package_type, self.author, self.name
+
+    def __hash__(self):
+        """Get the hash."""
+        return hash((self.package_type, self.public_id))
+
+    def __str__(self):
+        """Get the string representation."""
+        return "({package_type}, {public_id})".format(
+            package_type=self.package_type.value, public_id=self.public_id,
+        )
+
+    def __eq__(self, other):
+        """Compare with another object."""
+        return (
+            isinstance(other, PackageId)
+            and self.package_type == other.package_type
+            and self.public_id == other.public_id
+        )
+
+    def __lt__(self, other):
+        """Compare two public ids."""
+        return str(self) < str(other)
+
+
+class ComponentId(PackageId):
+    """
+    Class to represent a component identifier.
+
+    A component id is a package id, but excludes the case when the package is an agent.
+    >>> pacakge_id = PackageId(ConfigurationType.PROTOCOL, PublicId("author", "name", "0.1.0"))
+    >>> component_id = ComponentId(ComponentType.PROTOCOL, PublicId("author", "name", "0.1.0"))
+    >>> pacakge_id == component_id
+    True
+
+    >>> component_id2 = ComponentId(ComponentType.PROTOCOL, PublicId("author", "name", "0.1.1"))
+    >>> pacakge_id == component_id2
+    False
+    """
+
+    def __init__(self, component_type: Union[ComponentType, str], public_id: PublicId):
+        """
+        Initialize the component id.
+
+        :param component_type: the component type.
+        :param public_id: the public id.
+        """
+        component_type = ComponentType(component_type)
+        super().__init__(component_type.to_configuration_type(), public_id)
+
+    @property
+    def component_type(self) -> ComponentType:
+        """Get the component type."""
+        return ComponentType(self.package_type.value)
+
+    @property
+    def component_prefix(self) -> Tuple[ComponentType, str, str]:
+        """Get the component identifier without the version."""
+        package_prefix = super().package_prefix
+        package_type, author, name = package_prefix
+        return ComponentType(package_type.value), author, name
+
+    @property
+    def prefix_import_path(self) -> str:
+        """Get the prefix import path for this component."""
+        return "packages.{}.{}.{}".format(
+            self.public_id.author, self.component_type.to_plural(), self.public_id.name
+        )
+
+
 ProtocolId = PublicId
 SkillId = PublicId
 ContractId = PublicId
 
 
 class PackageConfiguration(Configuration, ABC):
-    """This class represent a package configuration."""
+    """
+    This class represent a package configuration.
+
+    A package can be one of:
+    - agents
+    - protocols
+    - connections
+    - skills
+    - contracts
+    """
+
+    default_configuration_filename: str
```

Comment:
the default configuration file name is now a class attribute of a package configuration class.